client/web/src/global/CoolCodeIntel.tsx

```diff
@@ -0,0 +1,842 @@
+import classNames from 'classnames'
+import * as H from 'history'
+import CloseIcon from 'mdi-react/CloseIcon'
+import MenuDownIcon from 'mdi-react/MenuDownIcon'
+import MenuUpIcon from 'mdi-react/MenuUpIcon'
+import OpenInAppIcon from 'mdi-react/OpenInAppIcon'
+import React, { useCallback, useEffect, useMemo, useState } from 'react'
+import { useHistory, useLocation } from 'react-router'
+import { Collapse } from 'reactstrap'
+
+import { HoveredToken } from '@sourcegraph/codeintellify'
+import { isErrorLike } from '@sourcegraph/common'
+import { useQuery } from '@sourcegraph/http-client'
+import { Markdown } from '@sourcegraph/shared/src/components/Markdown'
+import { displayRepoName } from '@sourcegraph/shared/src/components/RepoFileLink'
+import { Resizable } from '@sourcegraph/shared/src/components/Resizable'
+import { SettingsCascadeOrError } from '@sourcegraph/shared/src/settings/settings'
+import { renderMarkdown } from '@sourcegraph/shared/src/util/markdown'
+import {
+    RepoSpec,
+    RevisionSpec,
+    FileSpec,
+    ResolvedRevisionSpec,
+    toPositionOrRangeQueryParameter,
+    appendLineRangeQueryParameter,
+    appendSubtreeQueryParameter,
+    parseQueryAndHash,
+    formatSearchParameters,
+    addLineRangeQueryParameter,
+    lprToRange,
+} from '@sourcegraph/shared/src/util/url'
+import {
+    Tab,
+    TabList,
+    TabPanel,
+    TabPanels,
+    Tabs,
+    Link,
+    LoadingSpinner,
+    useLocalStorage,
+    CardHeader,
+    useDebounce,
+    Button,
+    useObservable,
+} from '@sourcegraph/wildcard'
+
+import {
+    CoolCodeIntelHighlightedBlobResult,
+    CoolCodeIntelHighlightedBlobVariables,
+    CoolCodeIntelReferencesResult,
+    CoolCodeIntelReferencesVariables,
+    LocationFields,
+    Maybe,
+} from '../graphql-operations'
+import { resolveRevision } from '../repo/backend'
+import { Blob, BlobProps } from '../repo/blob/Blob'
+import { parseBrowserRepoURL } from '../util/url'
+
+import styles from './CoolCodeIntel.module.scss'
+import { FETCH_HIGHLIGHTED_BLOB, FETCH_REFERENCES_QUERY } from './CoolCodeIntelQueries'
+
+export interface GlobalCoolCodeIntelProps {
+    coolCodeIntelEnabled: boolean
+    onTokenClick?: (clickedToken: CoolClickedToken) => void
+}
+
+export type CoolClickedToken = HoveredToken & RepoSpec & RevisionSpec & FileSpec & ResolvedRevisionSpec
+
+interface CoolCodeIntelProps extends Omit<BlobProps, 'className' | 'wrapCode' | 'blobInfo' | 'disableStatusBar'> {
+    clickedToken?: CoolClickedToken
+}
+
+export const isCoolCodeIntelEnabled = (settingsCascade: SettingsCascadeOrError): boolean =>
+    !isErrorLike(settingsCascade.final) && settingsCascade.final?.experimentalFeatures?.coolCodeIntel !== false
+
+export const CoolCodeIntel: React.FunctionComponent<CoolCodeIntelProps & { onClose: () => void }> = props => {
+    if (!props.coolCodeIntelEnabled) {
+        return null
+    }
+
+    return <CoolCodeIntelResizablePanel {...props} />
```

Comment:
I think we should wrap this here in an error boundary (or below the x to close it, if we can encapsulate that well enough), so the page doesn't brick completely when there's a bug in this experimental panel. In other places we have those too, but for something that directly lives in `<Layout/ >` this would be covered.
```suggestion
    return <ErrorBoundary
        location={null}
        render={error => (
            <div>
                Something broke here. Call Thorsten at 0800-HORSE.
                <pre>{JSON.stringify(error)}</pre>
            </div>
        )}
    >
  <CoolCodeIntelResizablePanel {...props} />
    </ErrorBoundary>
```